src/Microsoft.Diagnostics.Runtime/src/Desktop/DesktopHeapType.cs

```diff
@@ -477,26 +477,24 @@ public override bool IsFinalizable
 
         public override bool GetFieldForOffset(int fieldOffset, bool inner, out ClrInstanceField childField, out int childFieldOffset)
         {
-            int ps = DesktopHeap.PointerSize;
-            int offset = fieldOffset;
-
             if (!IsArray)
             {
+                int offset = fieldOffset;
+
                 if (!inner)
-                    offset -= ps;
+                    offset -= DesktopHeap.PointerSize;
 
                 foreach (ClrInstanceField field in Fields)
                 {
-                    if (field.Offset <= offset)
-                    {
-                        int size = field.Size;
+                    if (field.ElementType == ClrElementType.Unknown)
+                        break;
 
-                        if (offset < field.Offset + size)
-                        {
-                            childField = field;
-                            childFieldOffset = offset - field.Offset;
-                            return true;
-                        }
+                    if (offset >= field.Offset && 
```

Comment:
Nitpick:  This is much easier to read as:  `if (field.Offset <= offset && offset < field.Offset + field.Size)`.  That follows the rest of ClrMD's convention as well.

Reply:
Updated.